### ISSUE TITLE
Unexpected mask.filter behavior

### ISSUE THREAD

**Issue by wmakeev** [nemtsov/json-mask#16]
```js
let masked = mask({ 
  a: [
    { b: { name: 'foo' } },
    { b: 1 },
    { b: function noop() {} },
    { b: 0 },
    { b: null },
    { b: undefined }
  ]
}, 'a/b/name')

assert.deepEqual(masked, {
  a: [
    { b: { name: 'foo' } },   // ok
    {},                       // why empty? should be { b: 1 }
    { b: { name: 'noop' } },  // we really want to dive in to function object?
    { b: 0 },                 // ok
    { b: null },              // ok
    {}                        // ok
  ]
})
```

**Comment by wmakeev** [comment]
I try to fix this and #11 issue. 
But this solution have some breaking changes and I describe it later in pull request.

**Comment by nemtsov** [comment]
@wmakeev `{ b: 1 }` is empty because it doesn't have `name` as the filter requests (`a/b/name`). Good question re the `function`, what's your opinion?
